libclc/ptx-nvidiacl/libspirv/math/native_exp2.cl

```diff
@@ -14,4 +14,41 @@
 #define __CLC_FUNCTION __spirv_ocl_native_exp2
 #define __CLC_BUILTIN __nv_exp2
 #define __CLC_BUILTIN_F __CLC_XCONCAT(__CLC_BUILTIN, f)
+
+#ifdef cl_khr_fp16
+#pragma OPENCL EXTENSION cl_khr_fp16 : enable
+
+int __clc_nvvm_reflect_arch();
+
+_CLC_DEF _CLC_OVERLOAD half __clc_native_exp2(half x) {
+  if (__clc_nvvm_reflect_arch() >= 750) {
+    return __nvvm_ex2_approx_f16(x);
+  } else {
+    float upcast = x;
+    return __spirv_ocl_native_exp2(upcast);
+  }
```

Comment:
https://llvm.org/docs/CodingStandards.html#don-t-use-else-after-a-return
Please, don't use else after return and apply to all functions.

Reply:
Sure! Thanks for pointing it out @bader.